### PR TITLE
Homework2: Sergey Streltsov

### PR DESCRIPTION
**Pull request by Mantikor**

### CODE REVIEW

**Review by belovmd — CHANGES_REQUESTED** [review]
1. wrong PR naming
2. Missed code for external sites tasks
3. no description in  2 files

**Comment by Mantikor** [comment]
Fixed

File: src/homework02_task4.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    2). The program should produce an answer in less than a few seconds.
+    """
+    if len(sys.argv) == 2:
+        in_file = sys.argv[1]
+    else:
+        print('no valid file name...')
+        sys.exit()
+
+    test_str = open(in_file).read()
+    test_list = [test_str[i] for i in range(len(test_str))]
```
Comment (belovmd) [inline src/homework02_task4.py:67]: `[char for char in test_str]`?

File: src/homework02_task4.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    k = 0
```
Comment (belovmd) [inline src/homework02_task4.py:68]: what is k?

File: src/homework02_task4.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+def snowden():
+    """Snowden.
+
+    The program should be executed by calling
+    'python snowden.py cipheredmessage.txt' where "cipheredmessage.txt"
+    is a file with a ciphered message sent by Bob.
+    The message consists of lowercase English letters and its length
+    is at most 100 000. Output the message after step
+    2). The program should produce an answer in less than a few seconds.
```
Comment (belovmd) [inline src/homework02_task4.py:58]: no decryption method put to docsting

File: src/homework02_task4.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        sys.exit()
+
+    test_str = open(in_file).read()
+    test_list = [test_str[i] for i in range(len(test_str))]
+    k = 0
+    while k < (len(test_list) - 1):
+        if test_list[k] == test_list[k + 1]:
+            test_list.pop(k)
+            test_list.pop(k)
+            if k != 0:
```
Comment (belovmd) [inline src/homework02_task4.py:73]: `if not k`?

File: src/homework02_task4.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def letter_change():
+    """Letter change.
+
+    In each word in the text change k-th letter to the
+    selected symbol. If k > len(word) - exchange not needed.
+    """
+    input_text = input('Input text: ')
+    position_index = input('Input index for change: ')
```
Comment (belovmd) [inline src/homework02_task4.py:34]: `int(input(...))`

**Review by belovmd — COMMENTED** [review]
6
1. overdue
2. few external tasks
3. issues with code readability

**Pull request closed without merging**